canal/sync.go

```diff
@@ -175,6 +150,64 @@ func (c *Canal) runSyncBinlog() error {
 	return nil
 }
 
+type node struct {
+	db    string
+	table string
+}
+
+func parseStmt(stmt ast.StmtNode) (ns []*node) {
+	switch t := stmt.(type) {
+	case *ast.RenameTableStmt:
+		for _, tableInfo := range t.TableToTables {
+			n := &node{
+				db:    tableInfo.OldTable.Schema.String(),
+				table: tableInfo.OldTable.Name.String(),
+			}
+			ns = append(ns, n)
+		}
+	case *ast.AlterTableStmt:
```

Comment:
still `AlterTableStmt` problem,  `AlterTableStmt` contains field `Specs` and `AlterTableRenameTable spec` stores table name in its `NewTable` field, you know?   you should fetch new table name from it I think

Reply:
i think what we need is just clear old table cache.  for new table, it will be fetched when table cache miss. i think it's  unneccessary to get new table name here.

Reply:
It is, but it's also relay on definition of `OnTableChanged`, would user want to receive new table like create table, rename table?  we should make it clear @siddontang

Reply:
There is another PR #358 do this, I think we can merge this at first, and then introduce #358